documentation/source/quickstart.rst

```diff
@@ -4,291 +4,188 @@
+The :term:`DUT` is instantiated as the toplevel in the simulator
+without any HDL wrapper code.
 
-.. _quickstart_creating_a_test:
+The test is written in Python.
 
-Creating a test
----------------
+Cocotb wraps your HDL top level with the handle you pass it,
+drives stimulus onto the DUT's inputs and monitors the outputs,
+all directly from Python.
+In this documentation,
+and most of the examples in the project,
+the passed handle is called ``dut``.
 
-The test is written in Python. Cocotb wraps your top level with the handle you
-pass it. In this documentation, and most of the examples in the project, that
-handle is ``dut``, but you can pass your own preferred name in instead. The
-handle is used in all Python files referencing your :term:`RTL` project. Assuming we
-have a toplevel port called ``clk`` we could create a test file containing the
-following:
+Let's create a test file ``test_my_design.py`` containing the following:
 
 .. code-block:: python3
 
+    # This is file test_my_design.py
+
     import cocotb
     from cocotb.triggers import Timer
 
     @cocotb.test()
     async def my_first_test(dut):
         """Try accessing the design."""
 
-        dut._log.info("Running test!")
+        dut._log.info("Running test...")
         for cycle in range(10):
             dut.clk <= 0
-            await Timer(1, units='ns')
+            await Timer(1, units="ns")
             dut.clk <= 1
-            await Timer(1, units='ns')
-        dut._log.info("Running test!")
+            await Timer(1, units="ns")
 
-This will drive a square wave clock onto the ``clk`` port of the toplevel.
+        dut._log.info("my_signal_1 is", dut.my_signal_1.value)
+        assert dut.my_signal_2.value[0] == 0, "my_signal_2[0] is not 0!"
 
+        dut._log.info("Running test...done")
 
-Accessing the design
---------------------
 
-When cocotb initializes it finds the top-level instantiation in the simulator
-and creates a handle called ``dut``. Top-level signals can be accessed using the
-"dot" notation used for accessing object attributes in Python. The same mechanism
-can be used to access signals inside the design.
+This will first drive 10 periods of a square wave clock onto a port ``clk`` of the toplevel.
+After this, the clock stops,
+the value of ``my_signal_1`` is printed,
+and the value of index ``0`` of ``my_signal_2`` is checked to be ``0``.
 
-.. code-block:: python3
+Things to note:
 
-    # Get a reference to the "clk" signal on the top-level
-    clk = dut.clk
+* Use the ``@cocotb.test()`` decorator to mark the test function to be run
+* Use ``<=`` to assign a value to a signal (alternatively, use ``.value =``)
+* Use ``.value`` to get a signal's current value
 
-    # Get a reference to a register "count"
-    # in a sub-block "inst_sub_block"
-    count = dut.inst_sub_block.count
+The test shown is running sequentially, from start to end.
+Lines with :keyword:`await` wait there until the simulator is returning control back to cocotb.
```

Comment:
```suggestion
Each :keyword:`await` expression suspends execution of the test until whatever event the test is waiting for occurs and the simulator returns control back to cocotb (see :ref:`simulator-triggers`).
```

Reply:
Oops, I remember you had suggested that before, must have forgotten it.